src/mgmt/srv_drpc.c

```diff
@@ -572,6 +572,134 @@ ds_mgmt_drpc_pool_destroy(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+
+	rc = uuid_parse(req->uuid, uuid);
+	if (rc != 0) {
+		D_ERROR("Couldn't parse '%s' to UUID\n", req->uuid);
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	new_prop = daos_prop_alloc(1);
+	if (new_prop == NULL) {
+		D_ERROR("Failed to allocate daos property\n");
+		D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	if (req->property_case != MGMT__POOL_SET_PROP_REQ__PROPERTY_NUMBER) {
+		D_ERROR("Pool property request must be numeric\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+	new_prop->dpp_entries[0].dpe_type = req->number;
+
+	switch (req->value_case) {
+	case MGMT__POOL_SET_PROP_REQ__VALUE_STRVAL:
+		D_ASPRINTF(out_str_val, "%s", req->strval);
+		if (out_str_val == NULL) {
+			D_GOTO(out, rc = -DER_NOMEM);
+		}
+		new_prop->dpp_entries[0].dpe_str = out_str_val;
+		break;
+	case MGMT__POOL_SET_PROP_REQ__VALUE_NUMVAL:
+		new_prop->dpp_entries[0].dpe_val = req->numval;
+		break;
+	default:
+		D_ERROR("Pool property request with no value (%d)\n",
+				req->value_case);
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	rc = ds_mgmt_pool_set_prop(uuid, new_prop, &result);
+	if (rc != 0) {
+		D_ERROR("Failed to set pool property on %s: %d\n",
+				req->uuid, rc);
+		goto out;
+	}
+
+	if (result == NULL) {
+		D_ERROR("Null set pool property response\n");
+		D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	if (result->dpp_nr != 1) {
+		D_ERROR("Expected 1 property in response (got %d)\n",
+			result->dpp_nr);
+		D_GOTO(out_result, rc = -DER_INVAL);
+	}
+
+	if (result->dpp_entries[0].dpe_type != req->number) {
+		D_ERROR("Property req/resp mismatch (%d != %d)",
+			result->dpp_entries[0].dpe_type, req->number);
+		D_GOTO(out_result, rc = -DER_INVAL);
+	}
+
+	resp.property_case = MGMT__POOL_SET_PROP_RESP__PROPERTY_NUMBER;
+	resp.number = result->dpp_entries[0].dpe_type;
+
+	switch (req->value_case) {
+	case MGMT__POOL_SET_PROP_REQ__VALUE_STRVAL:
+		if (result->dpp_entries[0].dpe_str == NULL)
+			D_GOTO(out_result, rc = -DER_INVAL);
+		D_ASPRINTF(resp.strval, "%s",
+			   result->dpp_entries[0].dpe_str);
+		if (resp.strval == NULL)
+			D_GOTO(out_result, rc = -DER_NOMEM);
+		resp.value_case = MGMT__POOL_SET_PROP_RESP__VALUE_STRVAL;
+		break;
+	case MGMT__POOL_SET_PROP_REQ__VALUE_NUMVAL:
+		resp.numval = result->dpp_entries[0].dpe_val;
+		resp.value_case = MGMT__POOL_SET_PROP_RESP__VALUE_NUMVAL;
+		break;
+	default:
+		D_ERROR("Pool property response with no value (%d)\n",
+				req->value_case);
+		D_GOTO(out_result, rc = -DER_INVAL);
+	}
+
+out_result:
+	daos_prop_free(result);
+
+out:
+	daos_prop_free(new_prop);
+	D_FREE(out_str_val);
+
+	resp.status = rc;
+	len         = mgmt__pool_set_prop_resp__get_packed_size(&resp);
+	D_ALLOC(body, len);
+	if (body == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_MARSHAL;
+		D_ERROR("Failed to allocate drpc response body\n");
+	} else {
+		mgmt__pool_set_prop_resp__pack(&resp, body);
+		drpc_resp->body.len  = len;
+		drpc_resp->body.data = body;
+	}
+
+	mgmt__pool_set_prop_req__free_unpacked(req, NULL);
```

Comment:
Memory leak - Need to free resp.strval after it's packed.

Reply:
d'oh, thanks